src/firebase_functions/db.py

```diff
@@ -0,0 +1,211 @@
+"""
+Module for Cloud Functions that are triggered by the Firebase Realtime Database.
+"""
+# pylint: disable=protected-access
+import dataclasses as _dataclass
+import functools as _functools
+import typing as _typing
+import datetime as _dt
+import firebase_functions.options as _options
+import firebase_functions.private.util as _util
+import firebase_functions.core as _core
+import cloudevents.http as _ce
+from firebase_functions.core import Change  # Exported for user ease of typing events
+
+_event_type_written = "google.firebase.database.ref.v1.written"
+_event_type_created = "google.firebase.database.ref.v1.created"
+_event_type_updated = "google.firebase.database.ref.v1.updated"
+_event_type_deleted = "google.firebase.database.ref.v1.deleted"
+
+
+@_dataclass.dataclass(frozen=True)
+class DatabaseEvent(_core.CloudEvent[_core.T]):
+    """
+    A CloudEvent that contains a DataSnapshot or a Change<DataSnapshot>.
+    """
+
+    firebase_database_host: str
+    """
+    The domain of the database instance.
+    """
+
+    instance: str
+    """
+    The instance ID portion of the fully qualified resource name.
+    """
+
+    reference: str
+    """
+    The database reference path.
+    """
+
+    location: str
+    """
+    The location of the database
+    """
+
+    params: dict[str, str]
+    """
+    An dict containing the values of the path patterns.
+    Only named capture groups are populated - {key}, {key=*}, {key=**}
+    """
+
+
+_E1 = DatabaseEvent[Change[_typing.Any | None]]
+_E2 = DatabaseEvent[_typing.Any | None]
+_C1 = _typing.Callable[[_E1], None]
+_C2 = _typing.Callable[[_E2], None]
+
+
+def _db_endpoint_handler(
+    func: _C1 | _C2,
+    raw: _ce.CloudEvent,
+) -> None:
+    event_attributes = raw._get_attributes()
+    event_data = raw._get_data()
+    # TODO Params are built locally via path pattern which is currently unimplemented
```

Comment:
```suggestion
    # TODO Params are built locally via path pattern which is currently unimplemented
```

Extracting path patterns into params isn't currently implemented, you can still define patterns in your ref or instance names and they work, just no automatic `params` dict generation yet. I will look into this after `https` support due to time constraints.